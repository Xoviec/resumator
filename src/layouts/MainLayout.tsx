import CssBaseline from "@mui/material/CssBaseline";
import makeStyles from "@mui/styles/makeStyles";
import { FunctionComponent } from "react";
import "../assets/css/global.css";
import { Nav } from "../components/layout/Nav";
import { colors } from "../config/theme";

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
  },
  spacer: theme.mixins.toolbar,
  content: {
    flexGrow: 1,
    backgroundColor: colors.background,
    padding: theme.spacing(3),
  },
}));

export const MainLayout: FunctionComponent = ({ children }) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <CssBaseline />
      <Nav />
      <main className={classes.content}>
        <div className={classes.spacer} />
        {children}
      </main>
    </div>
  );
};
