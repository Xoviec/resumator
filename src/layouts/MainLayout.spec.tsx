import { render } from "@testing-library/react";
import { MainLayout } from "./MainLayout";

jest.mock("../components/layout/Nav", () => ({
  Nav: () => <div data-testid="test-nav" />,
}));
jest.mock("../config/theme");

describe("LoginLayout", () => {
  it("renders correctly", () => {
    const { container } = render(<MainLayout />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders Nav", () => {
    const { getByTestId } = render(<MainLayout />);
    expect(getByTestId("test-nav")).toBeInTheDocument();
  });
});
