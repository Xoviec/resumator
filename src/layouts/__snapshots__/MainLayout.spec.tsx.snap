// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginLayout renders correctly 1`] = `
<div
  class="css-k008qs"
>
  <div
    data-testid="test-nav"
  />
  <main
    class="css-15dnej8"
  >
    <div
      class="css-0"
    />
  </main>
</div>
`;
